test/unit/math/prim/mat/fun/qr_thin_R_test.cpp

```diff
@@ -0,0 +1,36 @@
+#include <stan/math/prim/mat.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathMatrix, qr_thin_R) {
+  stan::math::matrix_d m0(0, 0);
+  stan::math::matrix_d m1(4, 2);
+  m1 << 1, 2, 3, 4, 5, 6, 7, 8;
```

Comment:
Every test is done with a matrix of these numbers. Maybe mix it up a little? How about -1, -2, -3, -4, -6, -5?